Taskfile.yml

```diff
@@ -50,6 +50,8 @@ tasks:
   init-clean:
     desc: "Clean all caches and re-install development tools"
     cmds:
+      - docker buildx prune --all
+      - docker system prune --all --volumes
```

Comment:
Oh, that's quite too much! But I guess, we don't store too many things, so it should be fast to recreate everything every time?

My biggest concern is that if apart from FerretDB images I have some other things too, they will be pruned as well. As a FerretDB developer, I'm ok with it, but as a community maintainer, I won't be happy if everything I have is deleted.

Reply:
> Oh, that's quite too much! 

😁 it helped in my case.

> But I guess, we don't store too many things, so it should be fast to recreate everything every time?

yes, it's fast enough.


> My biggest concern is that if apart from FerretDB images I have some other things too, they will be pruned as well.

Not silently. Only if you press `yes`.

> As a FerretDB developer, I'm ok with it, but as a community maintainer, I won't be happy if everything I have is deleted.

If you don't want to do that prunes, you can just answer No twice and the task continues to run without pruning docker volumes.

Reply:
We already have "go clean -cache -testcache -modcache -fuzzcache" that is as destructive as those prune commands. `init-clean` is a not an every day function, and it is not even documented in out contributing guidelines. That's a tool of last resort that helps in complicated cases

Reply:
@AlekSi `go clean` won't delete my Docker volumes where I might store something totally irrelevant to FerretDB or even Go but really needed for something else.

Reply:
(and yes, "go clean" does not ask for confirmation, so it is even more destructive)

Reply:
> @AlekSi `go clean` won't delete my Docker volumes where I might store something totally irrelevant to FerretDB or even Go but really needed for something else.

WDYT of extracting docker prune commands to totally new task target? (please suggest target name).

Reply:
> go clean won't delete my Docker volumes where I might store something totally irrelevant to FerretDB or even Go

Yes, but `go clean` would prune your Go modules cache that might contain modules that are not recoverable. It also cleans fuzzing cache that might contain entries that are not recoverable too. Both commands can cause irrecoverable data loss

Reply:
I think we could make task description more scary :)

Reply:
A scare task description sounds enough to me, yes!

Reply:
Well, I hope the description of what this command now does is scary enough for you :imp: